src/test/regress/expected/pg13.out

```diff
@@ -170,15 +170,103 @@ SELECT create_distributed_table('test_table', 'a');
+NOTICE:  Copying data from local table...
+NOTICE:  copying the data has completed
+DETAIL:  The local data in the table is no longer visible, but is still on disk.
+HINT:  To remove the local data, run: SELECT truncate_local_data_after_distributing_table($$test_pg13.test_wal$$)
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+EXPLAIN (ANALYZE TRUE, WAL TRUE, COSTS FALSE, SUMMARY FALSE, BUFFERS FALSE, TIMING FALSE)
+INSERT INTO test_wal VALUES(2,22);
+                          QUERY PLAN
+---------------------------------------------------------------------
+ Custom Scan (Citus Adaptive) (actual rows=0 loops=1)
+   Task Count: 1
+   Tasks Shown: All
+   ->  Task
+         Node: host=localhost port=xxxxx dbname=regression
+         ->  Insert on test_wal_65013 (actual rows=0 loops=1)
+               WAL: records=1 bytes=63
+               ->  Result (actual rows=1 loops=1)
+(8 rows)
+
+-- Test WAL working for multi-shard query
+SET citus.explain_all_tasks TO on;
+EXPLAIN (ANALYZE TRUE, WAL TRUE, COSTS FALSE, SUMMARY FALSE, BUFFERS FALSE, TIMING FALSE)
+DELETE FROM test_wal RETURNING *;
+                                  QUERY PLAN
+---------------------------------------------------------------------
+ Custom Scan (Citus Adaptive) (actual rows=2 loops=1)
+   Task Count: 2
+   Tuple data received from nodes: 6 bytes
+   Tasks Shown: All
+   ->  Task
+         Tuple data received from node: 3 bytes
+         Node: host=localhost port=xxxxx dbname=regression
+         ->  Delete on test_wal_65012 test_wal (actual rows=1 loops=1)
+               WAL: records=1 bytes=54
+               ->  Seq Scan on test_wal_65012 test_wal (actual rows=1 loops=1)
+   ->  Task
+         Tuple data received from node: 3 bytes
+         Node: host=localhost port=xxxxx dbname=regression
+         ->  Delete on test_wal_65013 test_wal (actual rows=1 loops=1)
+               WAL: records=1 bytes=54
+               ->  Seq Scan on test_wal_65013 test_wal (actual rows=1 loops=1)
+(16 rows)
+
+-- insert items back for next query
+INSERT INTO test_wal VALUES(1,11), (2,22);
+-- make sure WAL works in distributed subplans
+EXPLAIN (ANALYZE TRUE, WAL TRUE, COSTS FALSE, SUMMARY FALSE, BUFFERS FALSE, TIMING FALSE)
+WITH cte_1 AS (DELETE FROM test_wal RETURNING *)
+SELECT * FROM cte_1;
+                                            QUERY PLAN
+---------------------------------------------------------------------
+ Custom Scan (Citus Adaptive) (actual rows=2 loops=1)
+   ->  Distributed Subplan XXX_1
+         Intermediate Data Size: 36 bytes
+         Result destination: Write locally
+         ->  Custom Scan (Citus Adaptive) (actual rows=0 loops=1)
+               Task Count: 2
+               Tuple data received from nodes: 0 bytes
+               Tasks Shown: All
+               ->  Task
+                     Tuple data received from node: 0 bytes
+                     Node: host=localhost port=xxxxx dbname=regression
```

Comment:
These two tasks should contain the line:
```
                WAL: records=1 bytes=54 
```
Like in the output above.

Reply:
strictly speaking, they shouldn't, since there are 0 rows deleted, but maybe the intention was to delete > 0 rows.

Reply:
opened an issue https://github.com/citusdata/citus/issues/4212